get-started/commit-code-and-submit-a-pull-request.md

```diff
@@ -1,11 +1,104 @@
-* what content needs to be filled in the pull request description
-* how to request pull request review from reviewers
-* how to get notified when the pull request received review comments
+The goal of this page is to cover some of the most common questions and problems that new contributors might face. Although this page covers some Git basics too, if you find that the content on this page is a little difficult to understand, it is recommended that you read some introductions to Git first as follows: 
 
+- Beginner and Getting started sections of [this tutorial] (https://www.atlassian.com/git/tutorials) from Atlassian 
+- [Documentation](https://docs.github.com/en/github/getting-started-with-github/set-up-git) and [guides](https://guides.github.com/introduction/git-handbook/) for beginners from Github
+- More in-depth [book](https://git-scm.com/book/en/v2/) from Git
+
+## Prerequisites
+
+Before creating a pull request, you need to make sure that you've installed Git, forked [pingcap/tidb](https://github.com/pingcap/tidb), and cloned the upstream repo to your PC. The instructions on this page use the command line interface to interact with Git; there are also several GUIs and IDE integrations that can interact with Git too.
+
+If you've cloned the upstream repo, you will be able to reference it using `origin` in your local repo. Next, you need to set up a remote for the repo your forked using the following command:
+
+```bash
+git remote add dev https://github.com/username/tidb.git
+```
+
+You can check the remote setting using the following command:
+
+```bash
+git remote -v
+# dev    https://github.com/username/tidb.git (fetch)
+# dev    https://github.com/username/tidb.git (push)
+# origin    https://github.com/pingcap/tidb.git (fetch)
+# origin    https://github.com/pingcap/tidb.git (push)
+```
+
+## Standard Process
+
+The following is a normal procedure that you're likely to use for the most common minor changes and PRs:
+
+1. Ensure that you're making your changes on top of master: `git checkout master`.
+2. Get the latest changes from remote: git pull master.
+3. Make a new branch for your changes: `git checkout -b issue-12345-fix`.
+4. Make some changes to the repo and test them.
+5. Stage your changes via `git add .` and then commit them with `git commit`. Making intermediate commits might be a good idea as well. Pay attention on unintentionally commit changes that should not be committed via `git add .`. You can use `git status` to check if the change set is correct.
+6. Push your changes to your fork: `git push --set-upstream dev issue-12345-fix`.
+7. Make a PR from your fork to the master branch of pingcap/tidb. For more information on how to make a PR, see [Making a Pull Request](https://guides.github.com/activities/forking/#making-a-pull-request) in GitHub Guides.
+
+When making a PR, take a look at the PR template and follow the commit message format, PR title format and checklists.
```

Comment:
it's better to illustrate more about the PR title and commit message:

- what's the format
- how to decide the package part
- some good and bad examples

Reply:
Maybe we add a link from "PR template" to https://raw.githubusercontent.com/pingcap/tidb/master/.github/pull_request_template.md?